core/ast2100/ast2100.js

```diff
@@ -0,0 +1,516 @@
+"use strict";
+/*global DCTSIZE, DCTSIZE2, JpegHuffmanTable, BitStream, AST2100IDCT, AAN_IDCT_SCALING_FACTORS, ZIGZAG_ORDER */
+/*global ATEN_QT_LUMA, ATEN_QT_CHROMA, TABLE_CLASS_AC, TABLE_CLASS_DC */
+/*global BITS_AC_LUMA, BITS_AC_CHROMA, BITS_DC_LUMA, BITS_DC_CHROMA, HUFFVAL_AC_LUMA, HUFFVAL_AC_CHROMA, HUFFVAL_DC_LUMA, HUFFVAL_DC_CHROMA */
+/*global inRangeIncl, isEmpty */
+/*global fmt_u8a, fmt_u8, fmt_u16 */
```

Comment:
I'm not a huge fan of JavaScript's scoping myself, so I have no really good answers except that your work might first need more module cleanup in the core files. :)

But until then you could try hiding everything under as few top level objects as possible. Ideally under the RFB object as it all is protocol stuff.